### DISCUSSION TITLE
Aspire.Pomelo.EntityFrameworkCore.MySql with WebApplication

### DISCUSSION THREAD

**Discussion by Lulalaby** [microsoft/aspire#5585] [Q&A] [answered]
I try building a api service, which is a WebApplication instead of a HostApplication.
The webapplication is in the started project from aspire.
I looked into using Aspire.Pomelo.EntityFrameworkCore.MySql, but the only function to AddMySqlDbContext only works for IHostApplicationBuilder not WebApplication, which is IApplicationBuilder.

Did I missed anything?
I don't really want to set up another host just for this.
i'd prefer if i can just inject it in the apphost through:
```cs
var cache = builder.AddRedis("cache");
var mysql = builder.AddMySql("mysql");
var mysqldb = mysql.AddDatabase("moderations");

var apiService = builder.AddProject<ApiService>("apiservice")
	.WithReference(mysqldb);
```

**Comment by davidfowl** [comment]
Can you share the entire project on github. `WebApplicationBuilder` implements `IHostApplicationBuilder`. It's unclear from the code sample above what the problem is. It seems like you're talking about the web application and not the apphost project, yet the code snippet shows the apphost project.

**Reply by Lulalaby** [reply]
Sure! I uploaded it here https://github.com/Lulalaby/Aspire-Test (without any sensitive data)
I wanted to use `Aspire.Pomelo.EntityFrameworkCore.MySql` on https://github.com/Lulalaby/Aspire-Test/tree/main/AITSYS.MiaModeration.ApiService

The relevant part where i use the WebApplicationBuilder is https://github.com/Lulalaby/Aspire-Test/blob/main/AITSYS.MiaModeration.ApiService/Program.cs#L7.

I'm just trying to get familiar with aspire right now, so I used `Pomelo.EntityFrameworkCore.MySql` temporary again, as can be seen here https://github.com/Lulalaby/Aspire-Test/blob/main/AITSYS.MiaModeration.ApiService/Program.cs#L13-L14

My goal was tho to use `Aspire.Pomelo.EntityFrameworkCore.MySql` since it is a aspire project, unless I'm mistaken here.

I'm not that familiar with aspire yet so I apologize for any logical errors in advance.

The project is using .NET 8
```pwsh
 ~#@❯ dotnet --list-sdks
6.0.425 [C:\Program Files\dotnet\sdk]
7.0.410 [C:\Program Files\dotnet\sdk]
8.0.401 [C:\Program Files\dotnet\sdk]
9.0.100-preview.7.24407.12 [C:\Program Files\dotnet\sdk]
```

I'm using Visual Studio 2022 Enterprise Preview (17.12.0 Preview 1.0), with any workload installed
With ReSharper Tools (2024.2.3) installed from Jetbrains 
OS: 
```
Edition	Windows 11 Enterprise
Version	23H2
Installed on	‎13/‎12/‎2023
OS build	22635.4145
Experience	Windows Feature Experience Pack 1000.22700.1039.0
```

**Reply by Lulalaby** [reply]
Also, what I've noticed, kinda weird behavior.
When I install `Aspire.Pomelo.EntityFrameworkCore.MySql` through NuGet, and use `AddMySqlDbContext` on the builder, it works (at least in the eyes of intellisense), till the point I build the project, then it errors out that no method could be found.

**Reply by Lulalaby** [reply]
Argh, GitHub desktop uploaded it as private again, sorry for that, switched the visibility to public

**Reply by davidfowl** [reply]
What should I do with this project to reproduce the problems you are running into?

PS: Did you install the aspire workload?

**Reply by Lulalaby** [reply]
Yes I have the aspire workload installed.

What I want is to get/use the, through the app host injected, mysql service or reference or whatever it's called in the web application (api) project.

As per
> WebApplicationBuilder implements IHostApplicationBuilder

It should just work with
```cs
var builder = WebApplication.CreateBuilder(args);
builder.AddMysqlDbContext<ModerationDbContext>("moderation");
```

But it doesn't

**Reply by davidfowl** [reply]
What happens when you do that?

**Reply by Lulalaby** [reply]
I'll try to get a debug log later or Monday and maybe a recording
It's kinda.. Weird

Thanks in advance

**Reply by Lulalaby** [reply, marked as answer]
Okay so apparently the workload was buggy, or ReSharper. I reinstalled both and the extension method is now correctly recognized on webapplication.

I don't know why it behaved "wrong" before, but I guess it's one of the "we'll leave it be" things.

Many thanks anyways, I wish you a nice rest of the weekend.
